packages/create-nx-workspace/bin/create-nx-workspace.ts

```diff
@@ -233,96 +160,12 @@ export const commandsObject: yargs.Argv<Arguments> = yargs
   ) as yargs.Argv<Arguments>;
 
 async function main(parsedArgs: yargs.Arguments<Arguments>) {
-  const {
-    name,
-    preset,
-    appName,
-    style,
-    standaloneApi,
-    routing,
-    nxCloud,
-    packageManager,
-    defaultBase,
-    ci,
-    skipGit,
-    commit,
-    framework,
-    docker,
-    bundler,
-  } = parsedArgs;
-
-  output.log({
-    title: `Nx is creating your v${nxVersion} workspace.`,
-    bodyLines: [
-      'To make sure the command works reliably in all environments, and that the preset is applied correctly,',
-      `Nx will run "${packageManager} install" several times. Please wait.`,
-    ],
-  });
-
-  const tmpDir = await createSandbox(packageManager);
-
-  const directory = await createApp(
-    tmpDir,
-    name,
-    packageManager as PackageManager,
-    {
-      ...parsedArgs,
-      preset,
-      appName,
-      style,
-      routing,
-      standaloneApi,
-      nxCloud,
-      defaultBase,
-      framework,
-      docker,
-      bundler,
-    }
-  );
-
-  if (!isKnownPreset(parsedArgs.preset)) {
-    await createPreset(parsedArgs, packageManager as PackageManager, directory);
-  }
-
-  let nxCloudInstallRes;
-  if (nxCloud) {
-    nxCloudInstallRes = await setupNxCloud(
-      name,
-      packageManager as PackageManager
-    );
-  }
-  if (ci) {
-    await setupCI(
-      name,
-      ci,
-      packageManager as PackageManager,
-      nxCloud && nxCloudInstallRes.code === 0
-    );
-  }
-  if (!skipGit) {
-    try {
-      await initializeGitRepo(directory, { defaultBase, commit });
-    } catch (e) {
-      output.error({
-        title: 'Could not initialize git repository',
-        bodyLines: [e.message],
-      });
-    }
-  }
-
-  showNxWarning(name);
-  pointToTutorialAndCourse(preset as Preset);
+  const { _, ...args } = parsedArgs;
+  await createWorkspace<Arguments>(args.preset, args);
 
-  if (nxCloud && nxCloudInstallRes.code === 0) {
-    printNxCloudSuccessMessage(nxCloudInstallRes.stdout);
+  if (isKnownPreset(args.preset)) {
+    pointToTutorialAndCourse(args.preset as Preset);
   }
-
-  await recordStat({
-    nxVersion,
-    command: 'create-nx-workspace',
-    useCloud: nxCloud,
-    meta: messages.codeOfSelectedPromptMessage('nxCloudCreation'),
-  });
 }
 
 async function getConfiguration(
```

Comment:
nit: can we rename this `normalizeArgsMiddleware` or similar? It wasn't clear how it was being used at first glance, and now is a good time to clean this stuff up.